Makefile

```diff
@@ -118,13 +118,23 @@ protogen: ## Generate Go code from .proto files
 	protoc --proto_path=./internal/proto --go_out=./internal --go-grpc_out=./internal ./internal/proto/*.proto
 	echo "✅ gRPC code generated."
 
-test: _check_ping_env ## Run all tests
-	@echo "  > Test: Running all Go tests..."
-	for dir in $(TEST_DIRS); do
-		echo "    -> $$dir"
-		$(GOTEST) $$dir
+test: test-auth _check_ping_env ## Run all tests (auth tests run first sequentially, then all others)
+	@echo "  > Test: Running all other Go tests..."
+	@for dir in $(TEST_DIRS); do \
+		if [[ "$$dir" != *"/auth" ]]; then \
+			echo "    -> $$dir"; \
+			$(GOTEST) $$dir; \
+		fi; \
 	done
-	echo "✅ All tests passed."
+	@echo "✅ All tests passed."
+
+test-auth: _check_ping_env ## Run auth tests sequentially (internal/auth then cmd/auth)
+	@echo "  > Test: Running auth tests sequentially to avoid shared resource conflicts..."
```

Comment:
Revisit: Is there any way we can make these not sequential? Will this make tests take a lot longer?